src/parser/jobs/drg/modules/Positionals.js

```diff
@@ -0,0 +1,244 @@
+}
+
+const TRUE_NORTH_CHARGES = 2
+const TRUE_NORTH_CD = ACTIONS.TRUE_NORTH.cooldown * 1000
+
+export default class Positionals extends Module {
+	static handle = 'positionals'
+	static title = t('drg.positionals.title')`Positionals`
+	static dependencies = [
+		'jumps',
+		'combatants',
+		'suggestions',
+	]
+
+	// tracking raiden thrust procs
+	_rtCombos = []
+	_currentCombo = null
+
+	// true north tracking
+	_tnCharges = 2
+	_nextChargeIn = null
+
+	constructor(...args) {
+		super(...args)
+
+		this.addHook('cast', {by: 'player', abilityId: ROTATION_IDS}, this._onGCD)
+		this.addHook('cast', {by: 'player', abilityId: PROC_IDS}, this._onProcGCD)
+		this.addHook('applybuff', {by: 'player', abilityId: STATUSES.RAIDEN_THRUST_READY.id}, this._procSuccess)
+		this.addHook('applybuff', {by: 'player', abilityId: STATUSES.TRUE_NORTH.id}, this._tnUsed)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onGCD(event) {
+		if (this._currentCombo) {
+			// did we use the proc? we might not due to invuln windows
+			const action = getDataBy(ACTIONS, 'id', event.ability.guid)
+			if (action.id === ACTIONS.RAIDEN_THRUST.id) {
+				this._rtCombos.used = true
+			}
+
+			// close the window
+			this._rtCombos.push(this._currentCombo)
+			this._currentCombo = null
+		}
+	}
+
+	_onProcGCD(event) {
+		// get action data
+		const action = getDataBy(ACTIONS, 'id', event.ability.guid)
+
+		if (!this._currentCombo) {
+			// if we don't have an open combo window, open one
+			this._currentCombo = {
+				next: NEXT_COMBO[action.id],
+				success: false,
+				used: false,
+				trueNorthCharges: this._tnCharges,
+				trueNorthCast1: this.combatants.selected.hasStatus(STATUSES.TRUE_NORTH.id),
+				time: event.timestamp,
+			}
+		} else if (this._currentCombo) {
+			// if the action isn't the next expected one, close the current window (not expecting a proc
+			// due to broken combo).
+			if (this._currentCombo.next !== action.id) {
+				// discard, let the broken combo module yell about this
+				this._currentCombo = null
+			} else {
+				// add the time
+				this._currentCombo.time = event.timestamp
+
+				// check for true north still active
+				this._currentCombo.trueNorthCast2 = this.combatants.selected.hasStatus(STATUSES.TRUE_NORTH.id)
+			}
+		}
+
+		this._updateTnCharges(event.timestamp)
+	}
+
+	_updateTnCharges(timestamp) {
+		// check to see if we're past the required cooldown time
+		if (this._nextChargeIn && this._nextChargeIn <= timestamp) {
+			this._tnCharges += 1
+
+			if (this._tnCharges < TRUE_NORTH_CHARGES) {
+				// if we're below the max, queue another charge since the cd will keep ticking
+				this._nextChargeIn += TRUE_NORTH_CD
+			} else {
+				// otherwise we're good, no recharge happening
+				this._nextChargeIn = null
+			}
+		}
+
+		// if we're over the max, uhhhhhhhhhh panic?
+		if (this._tnCharges > TRUE_NORTH_CHARGES) {
+			console.log('DRG Error: True north charges exceeded limit')
```

Comment:
This should be removed before it gets merged; we try to keep all `console.log()`s out of production. Same for the one a few lines down. I would recommend just bounding the charge count within [0,2] in this and the function below, rather than allowing them to get into weird states and logging.